File: DTOs/Sources/MeetingServiceDTOs/GetRecordDTO.swift
import Foundation

public struct GetRecordDTO: Codable {
    public var meetingId: UUID
    public var lang: String
    public var identity: GetIdentityDTO
    public var status: RecordStatus
    public var content: String
    public var attendancesAppendix: String
    public var votingResultsAppendix: String?
    public var iAmTheRecorder: Bool
    
    public init(meetingId: UUID, lang: String, identity: GetIdentityDTO, status: RecordStatus, content: String, attendancesAppendix: String, votingResultsAppendix: String? = nil, iAmTheRecorder: Bool = false) {
        self.meetingId = meetingId
        self.lang = lang
        self.identity = identity
        self.status = status
        self.content = content
        self.attendancesAppendix = attendancesAppendix
        self.votingResultsAppendix = votingResultsAppendix
        self.iAmTheRecorder = iAmTheRecorder
    }
}

public enum RecordStatus: String, Codable, CaseIterable {
    case underway
    case submitted
    case approved
}

File: android/app/src/main/java/net/ipv64/kivop/dtos/MeetingServiceDTOs/GetRecordDTO.kt
package net.ipv64.kivop.dtos.MeetingServiceDTOs

import java.util.UUID
import java.time.LocalDateTime

data class GetRecordDTO (
    var meetingId : UUID,
    var lang : String,
    var identity : GetIdentityDTO,
    var status : RecordStatus,
    var content : String,
    var attendancesAppendix : String,
    var votingResultsAppendix : String?,
    var iAmTheRecorder : Boolean,
)
enum class RecordStatus {
    underway,
    submitted,
    approved,
}
